### DISCUSSION TITLE
How to use rn-fitness-tracker

### DISCUSSION THREAD

**Discussion by raouf2004** [doville/rn-fitness-tracker#99] [Q&A] [answered]
Hi everyone, need help how can i use this it will be great if you share with me an example for daily Step counting, please.
Ps: iam new in RN please ur help will mean much for me .
thank you

**Comment by lukebars** [comment, marked as answer]
Hello, there is an example app in the repository. Also, there is a documentation for just that:
https://kilohealth.github.io/rn-fitness-tracker/docs/fundamentals/usage-instructions

**Comment by AdilHypeteq** [comment]
how to use FitnessTracker.getMaximumStepsForMinuteAndSixMinuteInterval() function?

**Reply by lukebars** [reply]
Hi @AdilHypeteq ! We might be wrong, yet as far as we know, there is no such function in this module

**Reply by alarm109** [reply]
What are you trying to achieve? Maybe we can suggest other method to achieve it

**Reply by AdilHypeteq** [reply]
In need to fetch and calculate the maximum steps for minute and six-minute intervals for the day

**Reply by alarm109** [reply]
Very interesting use case. Sorry to say, but we don't have this type of functionality. You can get steps by day or query total steps for time period. 

You could query steps for one minute, but doing this way for all day would be not optimal. Because you would need to do 24*60=1440 queries just for one day.
```ts
const permissions: AuthorizationPermissions = {
  healthReadPermissions: [HealthKitDataType.StepCount],
  googleFitReadPermissions: [GoogleFitDataType.Steps],
};

const getStepsForOneMinute = async () => {
  try {
    const authorized = await FitnessTracker.authorize(permissions);

    if (!authorized) return;

    const endDate = new Date();
    const startDate = new Date(endDate.getTime() - 1000 * 60);

    const steps = await FitnessTracker.queryTotal(
      FitnessDataType.Steps,
      startDate,
      endDate,
    );

    // returns the number of steps in one minute, e.g. 52
    console.log(steps);
  } catch (error) {
    // Handle error here
    console.log(error);
  }
};
```
